src/FakeItEasy/Configuration/PropertySetterConfiguration.cs

```diff
@@ -0,0 +1,92 @@
+namespace FakeItEasy.Configuration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using FakeItEasy.Core;
+
+    internal class PropertySetterConfiguration<TValue>
+        : IPropertySetterConfiguration<TValue>
+    {
+        private readonly MethodCallExpression setterExpression;
+
+        private readonly Func<LambdaExpression, IVoidArgumentValidationConfiguration>
+            voidArgumentValidationConfigurationFactory;
+
+        public PropertySetterConfiguration(
+            MethodCallExpression setterExpression,
+            Func<LambdaExpression, IVoidArgumentValidationConfiguration> voidArgumentValidationConfigurationFactory)
+        {
+            this.setterExpression = setterExpression;
+            this.voidArgumentValidationConfigurationFactory = voidArgumentValidationConfigurationFactory;
+        }
+
+        public IVoidArgumentValidationConfiguration To(TValue value)
+        {
+            Expression<Func<TValue>> valueExpression = () => value;
+            var newSetterExpression = this.CreateSetterExpressionWithNewValue(valueExpression);
+            return this.CreateArgumentValidationConfiguration(newSetterExpression);
+        }
+
+        public IAfterCallSpecifiedConfiguration Throws(Func<IFakeObjectCall, Exception> exceptionFactory)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Throws(exceptionFactory);
+        }
+
+        public IVoidConfiguration Invokes(Action<IFakeObjectCall> action)
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .Invokes(action);
+        }
+
+        public IAfterCallSpecifiedConfiguration CallsBaseMethod()
+        {
+            return this.CreateArgumentValidationConfiguration(this.setterExpression)
+                .CallsBaseMethod();
+        }
+
+        public IAfterCallSpecifiedConfiguration AssignsOutAndRefParametersLazily(

```

Comment:
I've removed `IVoidWithOutAndRefParametersConfiguration` from `IPropertySetterConfiguration` with no problems, so it will not appear until _after_ one uses `WhenArgumentsMatch` or similar. Continuing to track down remaining appearances.